src/main/scala/temple/generate/database/PostgresGenerator.scala

```diff
@@ -61,8 +72,16 @@ object PostgresGenerator extends DatabaseGenerator {
             .mkString(",\n")
             .pipe(indent(_))
         s"CREATE TABLE $tableName (\n$stringColumns\n);"
-      case Read(tableName, columns) =>
+      case Read(tableName, columns, condition) =>
+        val stringColumns = columns.map(_.name).mkString(", ")
+        val stringCondition = condition.map(generateCondition) match {
+          case Some(conditionString) => List(s"WHERE $conditionString")
+          case None                  => List()
+        }
+        (s"SELECT $stringColumns FROM $tableName" +: stringCondition).mkString("", " ", ";")
```

Comment:
Can we avoid the usage of `mkString` here? Maybe take it up to the `stringCondition` variable and just make this an interpolation line, since it's quite hard to read

Reply:
I’m really torn here, not sure of the most elegant way of representing this.

Reply:
Issue with this is that there needs to be a space between the $tablename and the WHERE, but there shouldn't be a space if there is no `stringCondition`. That's why I used the `List` and mkString, as it handles that nicely - I can't think of another nice way of doing all this without explicity adding the space at the beginning of the WHERE string

Reply:
Hmmm, feels like it should be an optional then, but then there's no mkString method